extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -621,36 +634,52 @@ public function formValidation() {
 				break;
 		}
 
-		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok' ;
+		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok';
 		$this->msg = $signupForm->msg;
 		$this->errParam = $signupForm->errParam;
 	}
 
 	/**
 	 * Disables User Signup Captcha for automated tests, mobile skin, and sites such as internal
-	 *
-	 * @throws MWException
 	 */
 	private function disableCaptcha() {
+		// Disable captcha for automated tests and wikia mobile and sites disabling it, e.g. Internal
+		if ( $this->shouldDisplayCaptcha() ) {
```

Comment:
Good point. This is somewhat of an interesting code flow with constructor calling this off the bat. To be cleaned in the next captcha refactoring round : )